### PR TITLE
Fix trailing slashes being kept in API initialization

### LINKED ISSUE #348: Server url needs normalized
Calling `jellyfin.createApi()` with a url that includes a trailing `/` leads to api calls being made with duplicate slashes (i.e. `https://example.com//system/info/public`)

### PR DESCRIPTION
**Pull request by ThibaultNocchi**
fix #348

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -32,7 +32,8 @@ export class Api {
 		accessToken = '',
 		axiosInstance: AxiosInstance = globalInstance
 	) {
-		this.basePath = basePath;
+		// Remove trailing slash if present
+		this.basePath = basePath.endsWith('/') ? basePath.slice(0, -1) : basePath;
```
Comment (thornbill) [inline src/api.ts:36]: What if it ends in multiple slashes?

Comment (ThibaultNocchi) [reply src/api.ts:36]: True, I could delete all of them. But don't you think there could be _some_ users with a weird setup and a JF instance ending with many slashes?

Comment (thornbill) [reply src/api.ts:36]: Good point... let's just stick with removing the last

**Review by thornbill — APPROVED** [review]

**Comment by JetpackDuba** [comment]
As a side note, until this fix is released in the next version, the following code will also fail:

```typescript
const candidates = await jellyfin.discovery.getAddressCandidates('https://yourdomain.com');
```

And this one too because it uses the same method `getAddressCandidates`:

```typescript
const servers = await jellyfin.discovery.getRecommendedServerCandidates('https://yourdomain.com');
```

It happens because when a `URL` is converted to a string using `toString()` it adds the trailing slash as done [here](https://github.com/jellyfin/jellyfin-sdk-typescript/blob/master/src/utils/address-candidates.ts#L84).

**Pull request merged**